candle-examples/examples/reinforcement-learning/ddpg.rs

```diff
@@ -99,50 +99,50 @@ impl ReplayBuffer {
     pub fn random_batch(
         &self,
         batch_size: usize,
-    ) -> Option<(Tensor, Tensor, Tensor, Tensor, Vec<bool>, Vec<bool>)> {
+    ) -> Result<Option<(Tensor, Tensor, Tensor, Tensor, Vec<bool>, Vec<bool>)>> {
         if self.size < batch_size {
-            None
+            Ok(None)
         } else {
-            let transitions: Vec<Transition> = thread_rng()
+            let transitions: Vec<&Transition> = thread_rng()
                 .sample_iter(Uniform::from(0..self.size))
                 .take(batch_size)
-                .map(|i| self.buffer.get(i).unwrap().clone())
+                .map(|i| self.buffer.get(i).unwrap())
                 .collect();
 
-            let states: Vec<Tensor> = transitions
+            let states: Result<Vec<Tensor>> = transitions
```

Comment:
No need for all the `Result` annotations like this, you can probably just use the `?` after `collect`.

Reply:
I tried that actually and I'm having trouble getting it to compile without these `Result` annotations..
First I'm told I need to specify `::<Vec<_>>` in each `collect()`, and then when I do that (which already doesn't reduce the amount of syntax) I get the following error:

```
error[E0277]: the `?` operator can only be applied to values that implement `Try`
   --> candle-examples/examples/reinforcement-learning/ddpg.rs:112:39
    |
112 |               let states: Vec<Tensor> = transitions
    |  _______________________________________^
113 | |                 .iter()
114 | |                 .map(|t| t.state.clone().unsqueeze(0))
115 | |                 .collect::<Vec<_>>()?;
    | |_____________________________________^ the `?` operator cannot be applied to type `Vec<Result<candle_core::Tensor, candle_core::Error>>`
    |
    = help: the trait `Try` is not implemented for `Vec<Result<candle_core::Tensor, candle_core::Error>>`
```

Reply:
You need to replace `collect::<Vec...>()?` with `collect::<Result...>()?` I think.

Reply:
Got it!